aptos-move/block-executor/src/view.rs

```diff
@@ -1107,7 +1175,7 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TResourceVi
     }
 
     fn resource_exists(&self, state_key: &Self::Key) -> anyhow::Result<bool> {
-        self.get_resource_state_value_impl(state_key, None, ReadKind::Exists)
+        self.get_resource_state_value_impl(state_key, UnknownOrLayout::Unknown, ReadKind::Exists)
```

Comment:
this is important - that we distinguish None from get_resource_state_value (which means no layout) and None from here - which meant we don't know if it has layout.